azurelinuxagent/common/exception.py

```diff
@@ -26,9 +26,10 @@ class ExitException(BaseException):
     """
     Used to exit the agent's process
     """
-    def __init__(self, reason):
+    def __init__(self, reason, add_event=True):
         super(ExitException, self).__init__()
         self.reason = reason
+        self.add_event = add_event
```

Comment:
it's odd to add this flag here... couldn't you use the reason instead, or derive a new exception?

Reply:
There was just one instance where it didnt make sense to add an event hence just added this flag. I can rewrite it as a new Exception or something no issues

Reply:
Done

Reply:
thanks